sw-emulator/lib/periph/src/soc_reg.rs

```diff
@@ -438,6 +444,12 @@ struct SocRegistersImpl {
     #[register_array(offset = 0x00c8, write_fn = on_write_generic_output_wires)]
     cptra_generic_output_wires: [u32; CPTRA_GENERIC_OUTPUT_WIRES_SIZE / 4],
 
+    #[register(offset = 0x00d0)]
+    cptra_hw_rev_id: u32,
+
+    #[register_array(offset = 0x00d4)]
+    cptra_fw_rev_id: [u32; CPTRA_FW_REV_ID_SIZE / 4],
```

Comment:
As mentioned above, we should make these read-only for the SoC

Reply:
I am not sure if we have that kind of support on the sw-emulator.

Reply:
Huh, you're right. We should probably add that at some point. But we can skip it for this PR.